springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/components/DefaultComponentsScanner.java

```diff
@@ -17,28 +24,79 @@
+            components = this.scanForAutoConfigurationComponents(basePackage);
+            components.addAll(this.scanForComponentsInConfiguration(basePackage, configurationBasePackage));
+
+        } else if (StringUtils.isNotBlank(configurationBasePackage)) {
+            log.debug("Scanning for component classes in configuration package {}", configurationBasePackage);
+            components = this.scanForComponentsInConfiguration(null, configurationBasePackage);
+
+        } else if (StringUtils.isNotBlank(basePackage)) {
+            log.debug("Scanning for component classes in {}", basePackage);
+            components = this.scanForAutoConfigurationComponents(basePackage);
+
+        } else {
+            throw new IllegalArgumentException("There must be either a basePackage or configurationBasePackage given");
+        }
 
+        return components;
+    }
+
+    private Set<Class<?>> scanForAutoConfigurationComponents(String basePackage) {
         ClassPathScanningCandidateComponentProvider provider = new ClassPathScanningCandidateComponentProvider(false);
         provider.addIncludeFilter(new AnnotationTypeFilter(Component.class));
+        provider.addExcludeFilter(new AnnotationTypeFilter(Configuration.class));
 
         return provider.findCandidateComponents(basePackage).stream()
-                .map(BeanDefinition::getBeanClassName)
-                .peek(className -> log.debug("Found candidate class: {}", className))
-                .map(this::getClass)
-                .filter(Optional::isPresent)
-                .map(Optional::get)
-                .collect(toSet());
+            .map(BeanDefinition::getBeanClassName)
+            .map(this::getClass)
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .collect(toSet());
+    }
+
+    private Set<Class<?>> scanForComponentsInConfiguration(String basePackage, String configurationBasePackage) {
+        final ClassPathScanningCandidateComponentProvider provider = new ClassPathScanningCandidateComponentProvider(false);
+        provider.addIncludeFilter(new AnnotationTypeFilter(Configuration.class));
+
+        return provider.findCandidateComponents(configurationBasePackage).stream()
+            .map(BeanDefinition::getBeanClassName)
+            .map(this::getClass)
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .map(configurationClass -> {
+
+                Set<Class<?>> beans = new LinkedHashSet<>();
+                for (final Method method : configurationClass.getDeclaredMethods()) {
+                    final Bean bean = AnnotationUtils.getAnnotation(method, Bean.class);
+                    if (bean != null) {
+                        final Class<?> returnType = method.getReturnType();
+                        final String beanPackage = returnType.getPackage().getName();
```

Comment:
As mentioned above, I think this would be too confusing, let's remove the package filtering logic here.